src/main/java/com/butbetter/storage/csv/CSVImportService.java

```diff
@@ -0,0 +1,75 @@
+package com.butbetter.storage.csv;
+
+import com.butbetter.storage.csv.Exceptions.FaultyCSVException;
+import com.butbetter.storage.file_upload.Exceptions.StorageException;
+import com.butbetter.storage.file_upload.Exceptions.StorageFileNotFoundException;
+import com.butbetter.storage.repository.FileProductRepository;
+import com.butbetter.storage.model.ProductInformation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.io.FileNotFoundException;
+import java.nio.file.Path;
+import java.util.List;
+
+/**
+ * File to CSV-Objects Converter Service
+ */
+@Service
+public class CSVImportService {
+
+	private final CSVConverter converter;
+	private final FileProductRepository repo;
+
+	private final Logger logger = LoggerFactory.getLogger(CSVImportService.class);
+
+	/**
+	 * Autowired CSVImportService Constructor
+	 *
+	 * @param converter Converter
+	 * @param repo      repo to save objects to
+	 */
+	@Autowired
+	public CSVImportService(CSVConverter converter, FileProductRepository repo) {
+		this.converter = converter;
+		this.repo = repo;
+	}
+
+	/**
+	 * Converting and then Saving all objects from the given File at Path
+	 *
+	 * @param path Path to the File with CSV Information
+	 *
+	 * @throws StorageFileNotFoundException thrown, if the File couldn't be properly stored/processed before
+	 * @throws FaultyCSVException thrown, if no convertable CSV Elements were found in the File
+	 */
+	public void fromFile(Path path) throws FaultyCSVException, StorageFileNotFoundException {
+		List<ProductInformation> info = getInformationOutOfFile(path);
+		if (info.isEmpty()) {
+			String message = "no elements where found in csv file at: " + path.toString();
+			logger.error(message);
+			throw new FaultyCSVException(message);
+		}
+
+		// TODO: repo needs to be added here
+		// repo.saveAll(info);
```

Comment:
Could you explain me the todo? Is the part of CSVImportService or do you  need a connection to the application service?

Reply:
The TODO is for the next comming pull request, which will link the Storage Service and needed Repos together
the Pull request for that will be ready to review approximately 3 Days after this one is pushed to stage